File: src/my_first_cpp_pkg/src/tests_interfaces/hardware_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "test_interfaces/msg/hardware_status.hpp"

class HardwareStatusPublisher : public rclcpp::Node
{
    public:
    HardwareStatusPublisher() : Node("hardware_status_publisher") 
    {
        m_timer = this->create_wall_timer(std::chrono::seconds(1), 
            std::bind(&HardwareStatusPublisher::timerCallback, this));
        m_publisher = this->create_publisher<test_interfaces::msg::HardwareStatus>("hardware_status", 10);
    }
    private:
    void timerCallback()
    {
        auto msg = test_interfaces::msg::HardwareStatus();
        msg.are_motors_ready = true;
        msg.debug_message = "Nothing to say.";
        msg.temperature = 47.3;
        m_publisher->publish(msg);
    }
    rclcpp::TimerBase::SharedPtr m_timer;
    rclcpp::Publisher<test_interfaces::msg::HardwareStatus>::SharedPtr m_publisher;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<HardwareStatusPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_first_package/my_first_package/tests_interfaces/hardware_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from test_interfaces.msg import HardwareStatus

class HardwareStatusPublisher(Node):
    def __init__(self, node_name):
        super().__init__(node_name)
        self.hw_status_pub_ = self.create_publisher(HardwareStatus, "hardware_status",10)
        self.create_timer(1.0, self.timer_callback)
        self.get_logger().info("Hardware status publisher started...")

    def timer_callback(self):
        msg = HardwareStatus()
        msg.are_motors_ready = True
        msg.temperature = 43.7
        msg.debug_message = "Nothing special"
        self.hw_status_pub_.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = HardwareStatusPublisher("hardware_status_publisher")
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()